src/GraphQL.Dynamic/LiteralGraphType/RemoteLiteralGraphType.cs

```diff
@@ -0,0 +1,344 @@
+
+                        var types = schema.Types
+                            .Where(typeFilter)
+                            .Select(schemaType =>
+                            {
+                                var typeName = schemaType.Name;
+
+                                // [RemoteLiteralGraphTypeMetadata(<remote.Moniker>, <remote.Url>, <typeName>)]
+                                // public class GeneratedRemoteLiteralGraphType : RemoteLiteralGraphType
+                                var typeBuilder = moduleBuilder.DefineType(typeName, TypeAttributes.Public | TypeAttributes.Class, typeof(RemoteLiteralGraphType));
+                                typeBuilder.SetCustomAttribute(new CustomAttributeBuilder(metadataAttributeConstructor, new[] { remote.Moniker, url, typeName }));
+
+                                // public GeneratedRemoteLiteralGraphType(): base(<remote>, <typeName>) { }
+                                var ctorBuilder = typeBuilder.DefineConstructor(MethodAttributes.Public, CallingConventions.Standard, Type.EmptyTypes);
+                                var ctorGenerator = ctorBuilder.GetILGenerator();
+                                ctorGenerator.Emit(OpCodes.Ldarg_0);
+                                ctorGenerator.Emit(OpCodes.Ldstr, url);
+                                ctorGenerator.Emit(OpCodes.Ldstr, typeName);
+                                ctorGenerator.Emit(OpCodes.Call, parentConstructor);
+                                ctorGenerator.Emit(OpCodes.Ret);
+
+                                return typeBuilder.CreateType();
+                            })
+                            .ToList();
+
+                        // Update cache
+                        var typeSet = new HashSet<Type>(types);
+                        RemoteServerTypes.AddOrUpdate(url, typeSet, (key, old) => typeSet);
+
+                        return types;
+                    });
+                })
+                .ToList();
+
+            // Wait for schema & type resolution
+            var jaggedTypes = await Task.WhenAll(tasks);
+
+            // Flatten
+            return jaggedTypes.SelectMany(t => t).ToList();
+        }
+
+        private static IEnumerable<FieldType> GetFieldsForFieldType(string remote, Introspection.TypeElement parentField)
+        {
+            FieldTypeResolver complexFieldTypeResolver = member =>
+            {
+                if (!(member is RemoteLiteralGraphTypeMemberInfo literalMember))
+                {
+                    return null;
+                }
+
+                if (!RemoteServerTypes.TryGetValue(remote, out var remoteTypes))
+                {
+                    return null;
+                }
+
+                var schemaType = remoteTypes.FirstOrDefault(t => t.Name == literalMember.TypeName);
+                var realType = literalMember.IsList
+                    ? typeof(ListGraphType<>).MakeGenericType(schemaType)
+                    : schemaType;
```

Comment:
this fails if the schemaType is a base GraphQL type e.g. a ListGraphType<StringGraphType>

Reply:
Again, great catch; this is fixed per my incoming fix commits